File: examples/node/effect-task.mjs
import { randomUUID } from "node:crypto";
import { BeatAPIClient } from "./lib/beatapi.mjs";

const effectId = process.env.BEATAPI_EFFECT_ID;
const imageUrl = process.env.BEATAPI_EFFECT_IMAGE_URL;
if (!effectId || !imageUrl) {
  throw new Error("Set BEATAPI_EFFECT_ID and BEATAPI_EFFECT_IMAGE_URL first.");
}

const client = new BeatAPIClient();
console.log(JSON.stringify(await client.getEffect(effectId), null, 2));
const task = await client.createEffectTask(
  { effect_id: effectId, images: [imageUrl] },
  { idempotencyKey: randomUUID() },
);
console.log(`Created ${task.id}`);
console.log(JSON.stringify(await client.waitForTask(task.id), null, 2));

File: examples/python/effect_task.py
import json
import os
import uuid

from beatapi import BeatAPIClient


effect_id = os.environ["BEATAPI_EFFECT_ID"]
image_url = os.environ["BEATAPI_EFFECT_IMAGE_URL"]
client = BeatAPIClient()
print(json.dumps(client.get_effect(effect_id), indent=2))
task = client.create_effect_task(
    {"effect_id": effect_id, "images": [image_url]},
    idempotency_key=str(uuid.uuid4()),
)
print(f"Created {task['id']}")
print(json.dumps(client.wait_for_task(task["id"]), indent=2))
